src/main/java/de/hsrm/vegetables/my_food_coop_service/controller/ReportsController.java

```diff
@@ -153,4 +165,72 @@
+        }
+
+        if (fromDate.isAfter(toDate)) {
+            throw new BadRequestError("fromDate cannot be after toDate", ErrorCode.TO_DATE_AFTER_FROM_DATE);
+        }
+
+        List<DisposedItem> disposedItems = getDisposedItems(fromDate, toDate);
+        DisposedItemList response = new DisposedItemList();
+        response.setItems(disposedItems);
+
+        VatDetailItem vatDetails = new VatDetailItem();
+        Float totalVat = disposedItems.stream()
+                .map(DisposedItem::getTotalVat)
+                .reduce(0f, Float::sum);
+        Float grossAmount = disposedItems.stream()
+                .map(DisposedItem::getGrossAmount)
+                .reduce(0f, Float::sum);
+
+        response.setGrossAmount(StockService.round(grossAmount, 2));
+        response.setTotalVat(StockService.round(totalVat, 2));
+        response.setVatDetails(DisposeService.getVatDetails(disposedItems));
+
+        return ResponseEntity.ok(response);
+    }
+
+    /**
+     * Find multiple disposes between Dates
+     *
+     * @param fromDate time window from offsetDateTime where item was disposed
+     * @param toDate   time window to offsetDateTime where item was disposed
+     * @return All disposes between fromDate and toDate
+     */
+    private List<DisposedItem> getDisposedItems(LocalDate fromDate, LocalDate toDate) {
+        // Local Dates only contain date information and are missing time information.
+        // Convert the LocalDate to a timestamp with the options specified below.
+        OffsetDateTime fromDateConverted = OffsetDateTime.of(fromDate, LocalTime.MIN, ZoneOffset.UTC);
+        OffsetDateTime toDateConverted = OffsetDateTime.of(toDate, LocalTime.MAX, ZoneOffset.UTC);
+
+        List<DisposedDto> itemsBetweenDates = disposedRepository.findAllByCreatedOnBetween(fromDateConverted, toDateConverted);
+        List<DisposedItem> disposedItems = new ArrayList<>();
+
+        itemsBetweenDates.forEach(disposedDto -> {
+            DisposedItem disposedItem = new DisposedItem();
+            disposedItem.setCreatedOn(disposedDto.getCreatedOn());
+            disposedItem.setAmount(disposedDto.getAmount());
+            disposedItem.setName(disposedDto.getStockDto().getName());
+            disposedItem.setUserId(disposedDto.getUserDto().getId());
+            disposedItem.setStockId(disposedDto.getStockDto().getId());
+            disposedItem.setPricePerUnit(disposedDto.getPricePerUnit());
+            disposedItem.setUnitType(disposedDto.getUnitType());
+            disposedItem.setVat(StockService.round(disposedDto.getVat(), 2));
+
+            float totalVat = StockService.round(disposedDto.getPricePerUnit() * disposedDto.getVat() * disposedDto.getAmount(), 2);
+            disposedItem.setTotalVat(totalVat);
+
+            disposedItem.setGrossAmount(StockService.round(totalVat + disposedDto.getPricePerUnit() * disposedDto.getAmount(), 2));
+            disposedItems.add(disposedItem);
```

Comment:
This function is already implemented here in the [StockMapper::disposedDtoToDisposedItem](https://github.com/HSRM-Vegetables/myfoodcoop-service/blob/8bac7ab97f71dc53f507091ec04d052c1fb575ca/src/main/java/de/hsrm/vegetables/my_food_coop_service/mapper/StockMapper.java#L45) and should be used / extended.

That also allows you to then write a function that maps `List<DisposedDto>` to `List<DisposedItem>` similar to this one: [StockMapper::listStockDtoToListStockResponse](https://github.com/HSRM-Vegetables/myfoodcoop-service/blob/8bac7ab97f71dc53f507091ec04d052c1fb575ca/src/main/java/de/hsrm/vegetables/my_food_coop_service/mapper/StockMapper.java#L38)